code/modules/fishing/fishing_gear.dm

```diff
@@ -33,6 +33,20 @@
 	proc/attackby_pre(source, atom/target, mob/user)
 		if (target && user && (src.last_fished < TIME + src.fishing_delay))
 			var/datum/fishing_spot/fishing_spot = null
+			if ((((istype(target, /obj/item/fishing_rod/upgraded)) && src.tier < 2))&& istype(source, /obj/item/fishing_rod/cybernetic))
+				src.tier = 2
+				src.icon = target.icon
+				src.icon_state = target.icon_state
+				user.visible_message("<span class='notice'>You upgrade your [source] with the Upgraded Fishing Rod</span>")
+				qdel(target)
+				return
+			if ((((istype(target, /obj/item/fishing_rod/master)) && src.tier < 3))&& istype(source, /obj/item/fishing_rod/cybernetic))
+				src.tier = 3
+				src.icon = target.icon
+				src.icon_state = target.icon_state
+				user.visible_message("<span class='notice'>You upgrade your [source] with the Master Fishing Rod</span>")
+				qdel(target)
+				return
```

Comment:
would it not make more sense to define this behaviour on the cybernetic rod itself?

Reply:
Given the structure of the rest of the fishing module it looked like "Keep the sub-items cleanish, put the logic in the root item" meant sense. I can make a proc in the root item that calls to the cybernetic fishing rod to do that logic before continuing on with the rest of the attackby_pre proc but would we really gain all that much?